mongo/crud_examples_test.go

```diff
@@ -0,0 +1,525 @@
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Printf("deleted document %v", deletedDocument)
+}
+
+func ExampleCollection_FindOneAndReplace() {
+	var coll *mongo.Collection
+
+	// find and replace at most one document in which the "name" field is "Bob" with {name: "Alice"}
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.FindOneAndReplace().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
+	replacement := bson.D{{"name", "Alice"}}
+	var replacedDocument bson.M
+	err := coll.FindOneAndReplace(context.TODO(), filter, replacement, opts).Decode(&replacedDocument)
+	if err != nil {
+		// ErrNoDocuments means that the filter did not match any documents in the collection
+		if err == mongo.ErrNoDocuments {
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Printf("replaced document %v", replacedDocument)
+}
+
+func ExampleCollection_FindOneAndUpdate() {
+	var coll *mongo.Collection
+
+	// find at most one document in which the "name" field is "Bob" and set the name to "Alice"
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.FindOneAndUpdate().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
+	update := bson.D{{"$set", bson.D{{"name", "Alice"}}}}
+	var updatedDocument bson.M
+	err := coll.FindOneAndUpdate(context.TODO(), filter, update, opts).Decode(&updatedDocument)
+	if err != nil {
+		// ErrNoDocuments means that the filter did not match any documents in the collection
+		if err == mongo.ErrNoDocuments {
+			return
+		}
+		log.Fatal(err)
+	}
+	fmt.Printf("updated document %v", updatedDocument)
+}
+
+func ExampleCollection_InsertMany() {
+	var coll *mongo.Collection
+
+	// insert documents {name: "Alice"} and {name: "Bob"}
+	// set the Ordered option to false to allow both operations to happen even if one of them errors
+	docs := []interface{}{
+		bson.D{{"name", "Alice"}},
+		bson.D{{"name", "Bob"}},
+	}
+	opts := options.InsertMany().SetOrdered(false)
+	res, err := coll.InsertMany(context.TODO(), docs, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("inserted documents with IDs %v\n", res.InsertedIDs)
+}
+
+func ExampleCollection_InsertOne() {
+	var coll *mongo.Collection
+
+	// insert the document {name: "Alice"}
+	// set the BypassDocumentValidation option to opt-out of document-level validation
+	opts := options.InsertOne().SetBypassDocumentValidation(true)
+	res, err := coll.InsertOne(context.TODO(), bson.D{{"name", "Alice"}}, opts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("inserted document with ID %v\n", res.InsertedID)
+}
+
+func ExampleCollection_ReplaceOne() {
+	var coll *mongo.Collection
+
+	// replace at most one document in which the "name" field is "Bob" with {name: "Alice"}
+	// specify the Upsert option to insert {name: "Alice"} if a document matching the filter isn't found
+	opts := options.Replace().SetUpsert(true)
+	filter := bson.D{{"name", "Bob"}}
```

Comment:
For clarity, filter on something that looks unique.

Reply:
I had a hard time coming up with a good example for this using a unique field. The shell examples use `name` too (https://docs.mongodb.com/manual/reference/method/db.collection.replaceOne/) so I decided to match.

Reply:
How about _id where the value is something that looks like a social security number (assuming you don't have to deal with creating an ObjectId)

Reply:
But why not ObjectId, as you do in other examples?

Reply:
This was changed to use `_id` in a previous commit. Maybe the comment was on an outdated copy?